source/Ship.h

```diff
@@ -61,12 +61,14 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 		~Bay() = default;
 		
 		// Copying a bay does not copy the ship inside it.
-		Bay(const Bay &b) : point(b.point), category(b.category), side(b.side), facing(b.facing), launchEffects(b.launchEffects) {}
+		Bay(const Bay &b) : point(b.point), category(b.category), side(b.side), facing(b.facing), launchEffects(b.launchEffects), retrieveEffects(b.retrieveEffects) {}
 		Bay &operator=(const Bay &b) { return *this = Bay(b); }
 		
 		Point point;
 		std::shared_ptr<Ship> ship;
 		std::string category;
+		// If this bay isn't reversed by a carry.
+		bool free = true;
```

Comment:
It feels a bit redundant to track the occupied/free state in a separate variable (but I also don't have an immediate suggestion on how to improve on this).

Reply:
We kinda have to. We need to have a way of saying: this bay is used by a ship but that ship isn't docked to it.

Reply:
I guess we could also ask the carried-ship if it is docked. (But performance wise we would then need to delay the bay-reservation until a ship actually wants to dock.)

Reply:
Meh. I could do it but I really don't think it's worth it. And also that bool is not *that* redundant imo :)